sperkovic

soperkov opened the pull request:
Rijesena vjezba 4

soperkov committed:
sperkovic
Rijesena vjezba 4

ncaklovic commented on app/app.h:
```diff
@@ -4,5 +4,58 @@
 {
+	class animal {
+	public:
+		virtual std::string species() = 0 {}
```
Ne treba vam implementacija _pure virtual_ funkcija (umjesto `{}` samo `;`)

ncaklovic commented on app/app.h:
```diff
@@ -4,5 +4,58 @@
+	class animal {
+	public:
+		virtual std::string species() = 0 {}
+		virtual uint32_t legs() = 0 {}
```
Obje funkcije mogu biti `const`

ncaklovic commented on app/app.h:
```diff
@@ -4,5 +4,58 @@
+		uint32_t lc = 0;
+	public:
+		std::string add_animal(animal*);
+		uint32_t legs();
```
I ovo može/treba biti `const`

ncaklovic commented on app/app.cpp:
```diff
@@ -3,5 +3,51 @@
+		case 3:
+			return std::make_unique<tarantula>();
+		default:
+			break;
```
Što će funkcija vratiti nakon `break`?

ncaklovic commented on test/test.cpp:
```diff
@@ -59,21 +59,21 @@ namespace all_tests
-			Assert::AreEqual("tarantula", lc.add_animal(animal_factory(3).get()).c_str());
-			Assert::AreEqual(16u, lc.legs());
-		}
+			TEST_METHOD(legg_counter_different_animals)
```
Nemojte mijenjati testove (ili barem nemojte _commit_-ati)

ncaklovic commented on app/main.cpp:
```diff
@@ -1,7 +1,28 @@
+			continue;
+		}
+
+		auto a = animal_factory(num);
```
Kažu i uputstva da za ovo koristite već napisane klase (`leg_counter`).

ncaklovic reviewed (commented):
Uključite [workflows](https://github.com/soperkov/v04/actions) na svom repo-u.

soperkov committed:
Izmjene
Ne znam što se commitalo u testovima, ali nisam ništa izmjenjivala u njima. Workflowse sam uključila još jučer

ncaklovic commented:
Još uvijek u programu nepotrebno dva puta kreirate životinju (zovete `animal_factory`).
Pa `add_animal` vraća string doslovno da biste ga tamo mogli ispisati.

Usput, `while(1)` u C++ je obično `while(true)`

soperkov committed:
izmjene

ncaklovic commented:
Sve u redu.
